index.js

```diff
@@ -54,6 +59,17 @@ module.exports = function * (next) {
     }
   }
 
+  //Adjust infinite end
+  if (end == Number.POSITIVE_INFINITY) {
```

Comment:
Thank you, and BTW, i found there are some parts might not be valid by jslint, could you fix those tough? :-)